include/deal.II/base/parameter_acceptor.h

```diff
@@ -592,12 +610,18 @@ class ParameterAcceptor : public Subscriptor
 
 private:
   /**
-   * A list containing all constructed classes of type
+   * A mutex to prevent writing on the class_list set from multiple threads.
+   */
+  static std::mutex class_list_mutex;
+
+  /**
+   * A set containing the address of all constructed classes of type
    * ParameterAcceptor.
    */
-  static std::vector<SmartPointer<ParameterAcceptor>> class_list;
+  static std::set<ParameterAcceptor *, internal::ParameterAcceptorCompare>
```

Comment:
Huh, fascinating that this works given that you're only forward-declaring `ParameterAcceptorCompare`!

Reply:
Out of curiosity, why does it make a difference whether you use a vector or a set?

Reply:
It doesn't actually make a difference, but the set reflects better the logic behind the class list (it really is an ordered set of objects, that may be destroyed in arbitrary ways).